packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-comment-box.tsx

```diff
@@ -93,14 +115,33 @@ export const PullRequestCommentBox = ({
   const [textSelection, setTextSelection] = useState({ start: 0, end: 0 })
   const dropZoneRef = useRef<HTMLDivElement>(null)
 
+  const [localPrincipalsMentionMap, setLocalPrincipalsMentionMap] = useState<PrincipalsMentionMap>({})
+
+  const {
+    principalsMentionMap,
```

Comment:
`principalsMentionMap` - A hash map that keeps tracking the user details that is being _mentioned_ in the comment box and it is required to populate the API payload and show the display name in preview instead showing email.